fail2ban/tests/clientreadertestcase.py

```diff
@@ -649,8 +649,9 @@ def testReadStockJailConfForceEnabled(self):
 
 			# and we know even some of them by heart
 			for j in ['sshd', 'recidive']:
-				# by default we have 'auto' backend ATM
-				self.assertTrue(['add', j, 'auto'] in comm_commands)
+				# by default we have 'auto' backend ATM, but some distributions can overwrite it, 
+				# (e.g. fedora default is 'systemd') therefore let check it without backend...
+				self.assertTrue([True for i in comm_commands if len(i) == 3 and i[0:2] == ['add', j]])
```

Comment:
Actually optimal line would be

any(len(i) == 3 and i[0:2] == ['add', j] for i in comm_commands)

! :-)



On July 27, 2016 9:45:51 AM EDT, "Serg G. Brester" <notifications@github.com> wrote:
>> @@ -649,8 +649,9 @@ def testReadStockJailConfForceEnabled(self):
>>  
>>  			# and we know even some of them by heart
>>  			for j in ['sshd', 'recidive']:
>> -				# by default we have 'auto' backend ATM
>> -				self.assertTrue(['add', j, 'auto'] in comm_commands)
>> +				# by default we have 'auto' backend ATM, but some distributions
>can overwrite it, 
>> +				# (e.g. fedora default is 'systemd') therefore let check it
>without backend...
>> +				self.assertTrue([True for i in comm_commands if len(i) == 3 and
>i[0:2] == ['add', j]])
>
>> performance difference is imho negligible
>
>Collection of 2000 negligible non performant differences ea. 1ms, makes
>the whole thing 2 seconds longer...
>Since then I've optimized our test-cases in 0.10, I respect it very
>strong, in particular by such easy-going things, that everyone can read
>(I'm about readability)...
>
>>  if you want the same -- make 2nd one a generator
>
>nope, this generator is larger, because contains more entries...
>
>Secondly that will be slower and slower with each extending of jail,
>regex of jail or filter, etc...
>
>
>---
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/fail2ban/fail2ban/pull/1490/files/595160d7f77e812b3ab6bcf5da4b58673d379c6e#r72441353